### PR TITLE
Jlopezbarb/manifestv2 refactor

### LINKED ISSUE #2111: Refactor the `GetManifestV2` logic
The logic of `GetManifestV2` should handle everything about reading manifest. This way, the caller doesn't need to know if it's reading an old manifest, a pipeline or a v2 manifest. The interface should be `GetManifestV2(path string)`, with `path` being optional. 

If `path` isn't specified, the logic should do something similar to the `GetManifest` function.

Also, all these functions shouldbn¡t be in the `context` package. They should be in the `model` package close to the manifestV2 struct definition.

We should also have unit tests for type of files and manifests.

### PR DESCRIPTION
**Pull request by jLopezbarb**
Fixes #2111

## Proposed changes
- Move the logic to read a manifest by a path from the context package to the model package
- Move the logic of inferring the manifest v2 to the model package

### CODE REVIEW

**Review by pchico83 — APPROVED** [review]

**Review by teresaromero — APPROVED** [review]

File: pkg/model/const.go [hunk 1]
```diff
@@ -304,6 +304,9 @@ const (
 	// BuildkitProgressEnvVar defines the output of buildkit
 	BuildkitProgressEnvVar = "BUILDKIT_PROGRESS"
 
+	// OktetoAppNameEnvVar defines the name of the app
+	OktetoAppNameEnvVar = "OKTETO_APP_NAME"
```
Comment (pchico83) [inline pkg/model/const.go:308]:
```suggestion
	OktetoNameEnvVar = "OKTETO_NAME"
```

File: pkg/model/manifest.go [hunk 2]
```diff
@@ -14,7 +14,88 @@
...
+//Archetype represents the type of manifest
+type Archetype string
+
+var (
+	//StackType represents a stack manifest type
+	StackType Archetype = "stack"
+	//OktetoType represents a okteto manifest type
+	OktetoType Archetype = "okteto"
+	//PipelineType represents a okteto pipeline manifest type
+	PipelineType Archetype = "pipeline"
```
Comment (pchico83) [inline pkg/model/manifest.go:38]: can we have one for the new manifest?

File: pkg/model/manifest.go [hunk 3]
```diff
@@ -56,8 +144,271 @@ type DeployInfo struct {
...
+//NewDeployInfo creates a deploy Info
 func NewDeployInfo() *DeployInfo {
 	return &DeployInfo{
-		Commands: make([]string, 0),
+		Commands: []string{},
+	}
+}
+
+//GetManifestV2 gets a manifest from a path or search for the files to generate it
+func GetManifestV2(manifestPath string) (*Manifest, error) {
```
Comment (pchico83) [inline pkg/model/manifest.go:155]: if `okteto.yml` is the new manifest, does it take precedence over the rest of manifests?

**Review by pchico83 — APPROVED** [review]

File: pkg/model/manifest.go (hunk 3 above)
Comment (jLopezbarb) [reply pkg/model/manifest.go:160]: Added

**Pull request merged**